# Installation Ansible sous debian 

Add the following line to /etc/apt/sources.list:
deb http://ppa.launchpad.net/ansible/ansible/ubuntu trusty main

$ sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 93C4A3FD7BB9C367
$ sudo apt update
$ sudo apt install ansible

https://docs.ansible.com/ansible/latest/installation_guide/index.html

# Inventory

"error Permission "
[SOLVED] error Permission user 

root@:/etc/ansible# ansible all -m ping
192.168.0.13 | UNREACHABLE! => {
    "changed": false, 
    "msg": "Failed to connect to the host via ssh: root@192.168.0.13: Permission denied (publickey,password).", 
    "unreachable": true
}
192.168.122.126 | UNREACHABLE! => {
    "changed": false, 
    "msg": "Failed to connect to the host via ssh: root@192.168.122.126: Permission denied (publickey,password).", 
    "unreachable": true
}

root@master:~# ssh-keygen -t rsa -C "name@example.org"

root@master:~# ssh-copy-id user@child1.dev
root@master:~# ssh-copy-id user@child2.dev


192.168.0.13 | SUCCESS => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python3"
    }, 
    "changed": false, 
    "ping": "pong"
}
[WARNING]: Platform linux on host 192.168.122.126 is using the discovered Python interpreter at /usr/bin/python, but future
installation of another Python interpreter could change this. See
https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information.
192.168.122.126 | SUCCESS => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    }, 
    "changed": false, 
    "ping": "pong"
}